execute/executetest/table.go

```diff
@@ -46,6 +46,11 @@ type Table struct {
 	Alloc *memory.Allocator
 	// IsDone indicates if this table has been used.
 	IsDone bool
+	// PartitionSource indicates which partition the table is on.
+	ResidesOnPartition int
+	// ParallelGroup is assigned during execution so we can see in the
+	// execution output which group the data originated from.
+	ParallelGroup int
```

Comment:
Is there another way to test out parallel execution of sources without adding these fields to `executetest.Table`? This class is used in many places (including outside this repo) and may also serve as an example of an implementation of `flux.Table` for some. In general a table does not need to know which partition its on (and it may even change partitions).

Reply:
I was looking for a way to verify that the data was indeed coming from multiple instantiations of the execution nodes, but couldn't think of any other way, other than modifying the data to include the parallel group. I'll keep thinking about it.

Reply:
I wonder if you could create a new implementation of `flux.Table` called `ParallelFromTable` (that embeds `executetest.Table`) and change the test `Source` to contain a slice of them.

Reply:
Yeah I could do that. I had thought cloning is not ideal, but if it's the preferred method then no problem.